sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,173 @@ def transform(self, X):
+
+        if not ((isinstance(self.sparse, six.string_types) and
+                self.sparse == "auto") or isinstance(self.sparse, bool)):
+            raise ValueError("sparse can only use be boolean or 'auto'"
+                             " got {0}".format(self.sparse))
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+        self.n_features_ = X.shape[1]
+
+        if self.features == "train":
+            _, self.feat_with_missing_ = self._get_missing_features_info(X)
+
+        return self
+
+    def transform(self, X):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : {array-like, sparse matrix}, shape = [n_samples, n_features]
+             The missing indicator for input data
+
+        """
+        if self.features == "train":
+            check_is_fitted(self, "feat_with_missing_")
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+        if X.shape[1] != self.n_features_:
+            raise ValueError("X has a different shape than during fitting.")
+
+        imputer_mask, feat_with_missing = self._get_missing_features_info(X)
+
+        if self.features == "train":
+            features = np.setdiff1d(feat_with_missing,
+                                    self.feat_with_missing_)
+            if features.size > 0:
+                warnings.warn("The features %s have missing values "
+                              "in transform but have no missing values "
+                              "in fit " % features, RuntimeWarning,
+                              stacklevel=1)
+            imputer_mask = imputer_mask[:, self.feat_with_missing_]
+
+        elif self.features == "all":
+            imputer_mask = imputer_mask
+
+        else:  # features is array-like
+            imputer_mask = imputer_mask[:, self.features]
+
+        return imputer_mask
+
+    def _get_missing_features_info(self, X):
+        if sparse.issparse(X) and self.missing_values != 0:
+            #  sparse matrix and missing values is not zero
+            imputer_mask = _get_mask(X.data, self.missing_values)
+            imputer_mask = X.__class__((imputer_mask, X.indices.copy(),
+                                        X.indptr.copy()), shape=X.shape,
+                                       dtype=X.dtype)
+            feat_with_missing = imputer_mask.sum(axis=0).nonzero()[1]
+            feat_with_missing = np.ravel(feat_with_missing)
+
+        else:
+            #  sparse with zero as missing value and dense matrix
+            if sparse.issparse(X):
+                X = X.toarray()
+            imputer_mask = _get_mask(X, self.missing_values)
+            imputer_mask = astype(imputer_mask, int, copy=False)
+            feat_with_missing = np.where(np.any(imputer_mask, axis=0))[0]
```

Comment:
this invocation of `where` is identical to `nonzero`. I'd rather you call it `nonzero` here for consistency with the sparse case above.